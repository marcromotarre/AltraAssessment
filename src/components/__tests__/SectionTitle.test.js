import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SectionTitle from '../SectionTitle';

/** this component should:
 *  - render section title with:
 *      - section name (section)
 *        - svg (icon)
 *        - title (label) by prop.name (required)
 *      - section number (label)  by prop.number (required)
 */

const props = {
  name: 'section name',
  number: 5,
};

describe('Section Title Component tests', () => {
  it('SectionTitle Snapshot', () => {
    const tree = renderer
      .create(
        <SectionTitle {...props}>
          <svg width={32}>text</svg>
        </SectionTitle>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('SectionTitle renders SectionName, SectionNumber and children', () => {
    const wrapper = shallow(
      <SectionTitle {...props}>
        <svg width={32}>text</svg>
      </SectionTitle>
    );

    expect(wrapper.children().length).toBe(2);
    const sectionName = wrapper.childAt(0);
    expect(sectionName.type().displayName).toBe('styled.section');
    expect(sectionName.children().length).toBe(2);
    expect(sectionName.childAt(0).type()).toBe('svg');
    expect(sectionName.childAt(0).prop('width')).toBe(32);
    expect(sectionName.childAt(0).text()).toBe(`text`);
    expect(sectionName.childAt(1).text()).toBe(`${props.name}`);
    const sectionNumber = wrapper.childAt(1);
    expect(sectionNumber.type().displayName).toBe('styled.label');
    expect(sectionNumber.children().length).toBe(1);
    expect(sectionNumber.childAt(0).text()).toBe(`${props.number}`);
  });
});
