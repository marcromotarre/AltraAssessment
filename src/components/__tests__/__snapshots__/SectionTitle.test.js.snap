// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Title Component tests SectionTitle Snapshot 1`] = `
<section
  className="sc-AxjAm eDLRgU"
>
  <section
    className="sc-AxirZ gXuGhx"
  >
    <svg
      width={32}
    >
      text
    </svg>
    <label>
      section name
    </label>
  </section>
  <label
    className="sc-AxiKw jbzapM"
  >
    <label>
      5
    </label>
  </label>
</section>
`;
